test/unit/loops.spec.ts

```diff
@@ -630,3 +630,27 @@ test("do...while double-negation", () => {
     expect(util.transpileString(code)).not.toMatch("not");
     expect(util.transpileAndExecute(code)).toBe(2);
 });
+
+test("for...in with pre-defined variable", () => {
+    util.testFunction`
+        const obj = { x: "y", foo: "bar" };
+
+        let x = "";
+        let result = [];
+        for (x in obj) {
+            result.push(x);
+        }
+        return result;
```

Comment:
Since objects are compared deeply these 2 tests may be merged to `return { x, result }`